core/trino-main/src/main/java/io/trino/execution/PrivilegeUtilities.java

```diff
@@ -44,6 +48,24 @@ public static Set<Privilege> parseStatementPrivileges(Node statement, Optional<L
         return privileges;
     }
 
+    public static Set<EntityPrivilege> parseEntityKindPrivileges(String entityKind, Metadata metadata, Optional<List<String>> privileges)
```

Comment:
This method is misnamed. There's no parsing involved here. I would rename it to "fetchEntityPrivileges" or similar.